src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -95,6 +95,23 @@ public void SetSection(ObjectNodeSection section)
             _currentSection = section;
         }
 
+        [DllImport(NativeObjectWriterFileName)]
+        private static extern void SetCodeSectionAttribute(IntPtr objWriter, string sectionName, CustomSectionAttributes attributes = 0, string comdatName = null);
+
+        public void SetCodeSectionAttribute(ObjectNodeSection section)
+        {
+            if (!section.IsStandardSection)
+            {
+                SetCodeSectionAttribute(_nativeObjectWriter, section.Name, GetCustomSectionAttributes(section), section.ComdatName);
+            }
+            else
+            {
+                SetCodeSectionAttribute(_nativeObjectWriter, section.Name);
+            }
+
+            _currentSection = section;
```

Comment:
Is this actually changing the current section being emitted? If not, we should not alter `_currentSection`